### PR TITLE
"Percent of parent" and "Percent of total" measure transformations

### LINKED ISSUE #16: Add "percent of parent" and "percent of root" transformations for measures
It should be possible to add measures which have additional, special transformations configured.

Percent of parent should present split values as percent of a value in the parent split.
Percent of parent should present split values as percent of a total value.

Transformations should be possible to configure with a "transform" attribute on a measure with allowed values of "percent-of-parent" and "percent-of-root".
```
- name: percent
  formula: $main.count()
  transform: percent-of-parent
```

### PR DESCRIPTION
**Pull request by piter75**
Fixes #16

### CODE REVIEW

File: src/client/components/dimension-tile/dimension-tile.tsx [hunk 1]
```diff
@@ -183,7 +183,7 @@ export class DimensionTile extends React.Component<DimensionTileProps, Dimension
     }
 
     if (sortOn.measure) {
-      query = query.performAction(sortOn.measure.toApplyExpression());
+      query = query.performAction(sortOn.measure.toApplyExpression(0));
```
Comment (pguzik) [inline src/client/components/dimension-tile/dimension-tile.tsx]: maybe this should be the default value 0 which is nonNested ?

File: src/common/models/essence/essence.ts [hunk 2]
```diff
@@ -584,11 +584,11 @@ export class Essence implements Instance<EssenceValue, EssenceJS> {
     return highlight.delta.getSingleClauseSet();
   }
 
-  public getApplyForSort(sort: SortExpression): ApplyExpression {
+  public getApplyForSort(sort: SortExpression, nestingLevel: number): ApplyExpression {
```
Comment (pguzik) [inline src/common/models/essence/essence.ts]: maybe 0 as non nested default value if possible ?

File: src/client/components/dimension-tile/dimension-tile.tsx (hunk 1 above)
Comment (piter75) [reply src/client/components/dimension-tile/dimension-tile.tsx]: Good idea!

File: src/common/models/essence/essence.ts (hunk 2 above)
Comment (piter75) [reply src/common/models/essence/essence.ts]: Good idea!

File: src/common/models/measure/measure.mocha.ts [hunk 3]
```diff
@@ -178,4 +197,58 @@ describe('Measure', () => {
...
+        const applyExpression = MeasureFixtures.noTransformationMeasure().toApplyExpression();
+        expect(applyExpression.toJS()).to.deep.equal(MeasureFixtures.applyWithNoTransformation());
+      });
+
+      it('creates simple formula expression at first level', () => {
+        const applyExpression = MeasureFixtures.noTransformationMeasure().toApplyExpression(1);
+        expect(applyExpression.toJS()).to.deep.equal(MeasureFixtures.applyWithNoTransformation());
+      });
+
+      it('creates simple formula expression at first level', () => {
```
Comment (mkuthan) [inline src/common/models/measure/measure.mocha.ts]: 99th level?

File: src/common/models/measure/measure.mocha.ts [hunk 4]
```diff
@@ -178,4 +197,58 @@ describe('Measure', () => {
     });
 
   });
+
+  describe('toApplyExpression', () => {
+
+    describe('no transformation', () => {
```
Comment (mkuthan) [inline src/common/models/measure/measure.mocha.ts]: could we use dynamically generated tests? all three tests are almost identical
https://mochajs.org/#dynamically-generating-tests

Comment (piter75) [reply src/common/models/measure/measure.mocha.ts]: Sure.
Funny thing I have been thinking about it when I was creating the tests ;-)

**Pull request merged**